src/components/Example/Example.tsx

```diff
@@ -1,23 +1,170 @@
+import algosdk from 'algosdk'
+import type _algosdk from 'algosdk'
+import { Buffer } from 'buffer'
+import { ICON as KMDICON } from '../../clients/kmd/constants'
 
 const getDynamicPeraWalletConnect = async () => {
   const PeraWalletConnect = (await import('@perawallet/connect')).PeraWalletConnect
   return PeraWalletConnect
 }
 
+class TestManualProvider implements CustomProvider {
+  algosdk: typeof _algosdk
+
+  constructor(algosdkStatic: typeof _algosdk) {
+    this.algosdk = algosdkStatic
+  }
+
+  async connect(metadata: Metadata) {
+    let address = prompt('Enter address of your account')
+    if (address && !this.algosdk.isValidAddress(address)) {
+      alert('Invalid address; please try again')
+      address = null
+    }
+    const authAddress = address
+      ? prompt("Enter address of the signing account; leave blank if account hasn't been rekeyed")
+      : undefined
+
+    return {
+      ...metadata,
+      accounts: address
+        ? [
+            {
+              address,
+              name: address,
+              providerId: PROVIDER_ID.CUSTOM,
+              authAddr: authAddress === null || authAddress === address ? undefined : authAddress
+            }
+          ]
+        : []
+    }
+  }
+
+  async disconnect() {
+    //
+  }
+
+  async reconnect(_metadata: Metadata) {
+    return null
+  }
+
+  async signTransactions(
+    connectedAccounts: string[],
+    txnGroups: Uint8Array[] | Uint8Array[][],
+    indexesToSign?: number[] | undefined,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _returnGroup?: boolean | undefined
+  ): Promise<Uint8Array[]> {
+    // If txnGroups is a nested array, flatten it
+    const transactions: Uint8Array[] = Array.isArray(txnGroups[0])
+      ? (txnGroups as Uint8Array[][]).flatMap((txn) => txn)
+      : (txnGroups as Uint8Array[])
+
+    // Decode the transactions to access their properties.
+    const decodedTxns = transactions.map((txn) => {
+      return this.algosdk.decodeObj(txn)
+    }) as Array<_algosdk.EncodedTransaction | _algosdk.EncodedSignedTransaction>
+
+    const signedTxns: Array<Uint8Array> = []
+
+    let idx = -1
+    for (const dtxn of decodedTxns) {
+      idx++
+      const isSigned = 'txn' in dtxn
+
+      // push the incoming txn into signed, we'll overwrite it later
+      signedTxns.push(transactions[idx])
+
+      // Its already signed, skip it
+      if (isSigned) {
+        continue
+        // Not specified in indexes to sign, skip it
+      } else if (indexesToSign && indexesToSign.length && !indexesToSign.includes(Number(idx))) {
+        continue
+      }
+      // Not to be signed by our signer, skip it
+      else if (!connectedAccounts.includes(this.algosdk.encodeAddress(dtxn.snd))) {
+        continue
+      }
+
+      const unsignedTxn = this.algosdk.decodeUnsignedTransaction(transactions[idx])
+
+      const forSigning = Buffer.from(
+        this.algosdk.encodeObj({
+          txn: unsignedTxn.get_obj_for_encoding()
```

Comment:
I had a look at `get_obj_for_encoding` and it looks to me that it doesn't spit out the transaction in a format that is ready to be encoded + hashed + signed other than in goal (i.e cold vault system, hsm, hashicorp vault, raw EdDSA tool, etc)

`get_obj_for_encoding`  returns the tx object with public keys instead of the addresses; which is what you would expect in a properly "ready-to-encode-and-sign" message. I think there's a `goal` bias here. 

Wondering if anyone has ran into issues related to this and if we should rely algosdk.

Reply:
In your implementation could you create the tx object using `get_obj_for_encoding()` then replace the public key with the address before encoding? `algosdk.encodeAddress(new Uint8Array(Buffer.from(tx.snd, 'base64')))`

Reply:
Would there be a way to satisfy both the `goal` and non-`goal` use cases? If not, as an example I'd imagine `goal` is the easiest way to demo the provider. An alternative solution could be added to the documentation.

Reply:
Sounds good! If you can provide an example that works for you, we can add a new Storybook story that demonstrates EdDSA signing.

Re: `bytesToSign()` though, doesn't that also rely on `get_obj_for_encoding()` and return the public key?

https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.ts#L1116

Reply:
Hey @drichar it does; but `bytesToSign()` knows how to perform all the rights tricks and conversions to spit out the raw data to perform EdDSA on. 

I think i found the **_one flow to conquer it all_** : 
- Return the result of **bytesToSign()** instead of **get_obj_for_encoding()** and update the instructions to use the `rawsend` method on `goal`. 

That should cover any situation wherre someone is signing with whatever (libsodium, tweenacl, HSM, Hashicorp Vault, Copper, anything that supports Ed25519 / EdDSA). 

Thoughts?

Reply:
CC: @robdmoore

Reply:
Love it! Thanks for investigating. Will make this change when I come back to the PR.

Reply:
Unfortunately, when trying this out it's not possible to get goal to rawsign, you can only rawsend, which can't work with this provider since we don't want goal to actually send the transaction :/

My suggestion would be that the wallet provider implementation that supports both has a tab/button to select between goal and non-goal signing and provides a different base64 to copy/paste.